### PR TITLE
Feature/ab2 d 4399 hpms v2

### PR DESCRIPTION
**Pull request by sb-dkniesteadt**
**JIRA Tickets:**

[AB2D-4399](https://jira.cms.gov/browse/AB2D-4399) - Description
 
### What Does This PR Do?
Adds hpms auth v2 to ab2d. Fixes test failing due to dates failing in dec.

### What Should Reviewers Watch For?
What might break due to changes. 

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [x] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [x] Code checked for PHI/PII exposure

### CODE REVIEW

File: worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesIntegrationTest.java [hunk 1]
```diff
@@ -279,7 +281,14 @@ void whenCoverageChangesAreSmall_passCoverageStableCheck() {
     @Test
     void whenCoverageSmallPercentage_passCoverageStableCheck() {
 
-        createCoveragePeriods();
+        //Won't work in March due to dec attestation.
+        if(ATTESTATION_TIME.getMonth().getValue() == 12)
```
Comment (nflearl) [inline worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesIntegrationTest.java:285]: formatting cleanup needed.

File: worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesUnitTest.java [hunk 2]
```diff
@@ -174,20 +173,26 @@ void whenCoverageUnstable_failCoverageStabilityCheck() {
         contract.setContractNumber("TEST");
         contract.setAttestedOn(ATTESTATION_TIME.toOffsetDateTime());
 
+        ZonedDateTime nonDecAttestationTime;
+        if(ATTESTATION_TIME.getMonth().getValue() == 12)
```
Comment (nflearl) [inline worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesUnitTest.java:177]: formatting cleanup needed.

File: worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesIntegrationTest.java [hunk 3]
```diff
@@ -279,7 +281,14 @@ void whenCoverageChangesAreSmall_passCoverageStableCheck() {
...
+        {
+            createCoveragePeriods(ATTESTATION_TIME.plusMonths(1));
+        }
+        else{
```
Comment (nflearl) [inline worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesIntegrationTest.java:289]: formatting cleanup needed.

File: worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesIntegrationTest.java [hunk 4]
```diff
@@ -233,7 +232,10 @@ void whenCoverageUnstable_failCoverageStabilityCheck() {
 
         assertEquals(expectedIssues, issues.size());
         issues.forEach(issue -> assertTrue(issue.contains("enrollment changed")));
-        assertTrue(issues.get(0).contains("20%"));
+
+        //won't work in March
+        if(attestationMonth.getMonth() != 12)
```
Comment (nflearl) [inline worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesIntegrationTest.java:237]: formatting cleanup needed.

**Review by nflearl — CHANGES_REQUESTED** [review]

File: worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesIntegrationTest.java [hunk 5]
```diff
@@ -233,7 +232,10 @@ void whenCoverageUnstable_failCoverageStabilityCheck() {
 
         assertEquals(expectedIssues, issues.size());
         issues.forEach(issue -> assertTrue(issue.contains("enrollment changed")));
-        assertTrue(issues.get(0).contains("20%"));
+
+        //won't work in March
+        if (attestationMonth.getMonth() != 12)
```
Comment (sharshar-sb) [inline worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesIntegrationTest.java:237]: Isn't it March?

**Review by nflearl — APPROVED** [review]
Please create a ticket for a revisit to these tests to make them more robust.

File: worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesIntegrationTest.java (hunk 5 above)
Comment (sb-dkniesteadt) [reply worker/src/test/java/gov/cms/ab2d/worker/processor/coverage/check/CoverageCheckPredicatesIntegrationTest.java:238]: Yes. Added comments clarifying code

**Review by sharshar-sb — APPROVED** [review]

**Pull request merged**